File: Src/Domain/Goal.cs
using System.ComponentModel.DataAnnotations;

namespace Domain;

/// <summary>
/// Goals that user has set
/// </summary>

public class Goal
{
    public Guid Id { get; set; }
    [Required]
    public string Name { get; set; }
    public string Description { get; set; }
    public DateTime EndDate { get; set; }

    public decimal CurrentAmount { get; set; }

    public decimal RequiredAmount { get; set; }

    public Guid BudgetId { get; set; }
    public Budget Budget { get; set; }

    public IEnumerable<Saving> Savings { get; set; } = new List<Saving>();
}

File: Src/WebUI/src/app/models/goal.ts
export interface Goal {
  id: string
  name: string
  description: string | null
  endDate : Date
  currentAmount : number
  requiredAmount : number
}

export class Goal implements Goal {
  constructor(init?: GoalFormValues) {
    Object.assign(this, init)
  }
}

export class GoalFormValues {
  id?: string = undefined
  name: string = ''
  description: string | null = null
  endDate : Date | null = null
  currentAmount : number = 0
  requiredAmount : number = 0
}